esp/services/ws_ecl/ws_ecl_service.cpp

```diff
@@ -214,8 +214,10 @@ bool CWsEclService::init(const char * name, const char * type, IPropertyTree * c
             daliAddress.append(*daliServers++);
     }
 
-    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory(true);
     Owned<IConstEnvironment> environment = factory->openEnvironment();
```

Comment:
Should this be throwing an exception inside openEnvironment instead of having many places test it.  If there are valid situations where it is ok to continue if non-null then there could be another parameter to indicate that.

Reply:
agree, in practice it's not possible for the openEnvironment implementation to return null anyway..

Reply:
@jakesmith I am not sure how to change the existing code to match with the new exception inside openEnvironment(). For example, in envGetConfigurationDirectory() (dalienv.cpp) called by queryDllServer(), if the openEnvironment() returns nullptr, a default value is used. If the openEnvironment() throws exception, should we catch the exception and use the default value?

Reply:
Might we do need a "throwExceptionIfCannotOpen" in the openEnvironment()?

Reply:
> if the openEnvironment() returns nullptr, a default value is used. If the openEnvironment() throws exception, should we catch the exception and use the default value?

No, openEnvironment() should never return null and it's impossible for the current implementation  to do so, so the 'if (env)' check inside envGetConfigurationDirectory().
As long as openEnvironment is guaranteed to return an IConstEnvironment OR throw an exception, no check after each call to openEnvironment() is necessary.

Reply:
@wangkx - back to you.